src/lib/caching/structures/guilds/GuildMember.ts

```diff
@@ -85,6 +85,75 @@ export class GuildMember extends Structure {
 		return this.client.users.get(this.id) ?? null;
 	}
 
+	/**
+	 * The calculated permissions from the member's {@link GuildMemberRoleStore roles}.
+	 * @since 0.0.1
+	 */
+	public get permissions(): Readonly<Permissions> {
+		if (this.id === this.guild.ownerID) return new Permissions(Permissions.ALL).freeze();
+
+		const permissions = new Permissions(this.roles.map(role => role.permissions));
+		return (permissions.has(Permissions.FLAGS.ADMINISTRATOR) ? permissions.add(Permissions.ALL) : permissions).freeze();
+	}
+
+	/**
+	 * Whether or not the {@link ClientUser client user} can manage this member. This is based on:
+	 * - The member is not the {@link Guild#owner guild owner}.
+	 * - The member is not the {@link ClientUser client user} itself.
+	 * - The {@link ClientUser client user} is the owner of the {@link Guild}.
+	 * - The {@link ClientUser client user}'s {@link GuildMemberRoleStore#highest highest role} is higher than the member's.
+	 * @since 0.0.1
+	 * @returns `true` when any of the conditions are met, `null` when the {@link ClientUser client user}'s member is not
+	 * cached (or when {@link Client#user} is null), or `false` otherwise.
+	 */
+	public get manageable(): boolean | null {
+		// If the user is the owner, it is not manageable.
+		if (this.id === this.guild.ownerID) return false;
+
+		// If the client user is not cached, return null.
+		const { user } = this.client;
+		if (!user) return null;
+
+		// If the user is the same as the client user, it is not manageable.
+		if (this.id === user.id) return false;
```

Comment:
How so? If I'm me, I can DEFINITELY manage myself?

Reply:
You can't kick nor ban yourself, that's what this check does.

Reply:
But manageable !== kickable nor bannable. Those checks go in their respective functions. If I'm me, I can manage myself (add roles, remove roles, change nickname) as long as I have the required permission(s)

Reply:
This is a design issue, I myself think that manageable should be a subset of kickable and bannable, but I want @bdistin's opinion on this too.

Reply:
Manageable means very different things to different people without a clear definition of what that is. While yes, adding roles/removing roles is a type of manageable, those require more permissions than what's defined in this getter.

Maybe make manageable protected _managable, and add in manageRoles and manageNickname (rolable and nickable sound weird, not that those others don't sound weird too).